xarray/core/missing.py

```diff
@@ -0,0 +1,258 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from collections import Iterable
+
+import numpy as np
+import pandas as pd
+
+
+from .computation import apply_ufunc
+
+
+class BaseInterpolator(object):
+    '''gerneric interpolator class for normalizing interpolation methods'''
+    cons_kwargs = {}
+    call_kwargs = {}
+    f = None
+    kind = None
+
+    def __init__(self, xi, yi, kind=None, **kwargs):
+        self.kind = kind
+        self.call_kwargs = kwargs
+
+    def __call__(self, x):
+        return self.f(x, **self.call_kwargs)
+
+    def __repr__(self):
+        return "{type}: kind={kind}".format(type=self.__class__.__name__,
+                                            kind=self.kind)
+
+
+class NumpyInterpolator(BaseInterpolator):
+    def __init__(self, xi, yi, kind='linear', fill_value=None, **kwargs):
+        self.kind = kind
+        self.f = np.interp
+        self.cons_kwargs = kwargs
+        self.call_kwargs = {'period': self.cons_kwargs.pop('period', None)}
+
+        self._xi = xi
+        self._yi = yi
+
+        if self.cons_kwargs:
+            raise ValueError(
+                'recieved invalid kwargs: %r' % self.cons_kwargs.keys())
+
+        if fill_value is None:
+            self._left = np.nan
+            self._right = yi[-1]
+        elif isinstance(fill_value, Iterable):
+            self._left = fill_value[0]
+            self._right = fill_value[1]
```

Comment:
If this only works for a pair of values, let's restrict this to `isintance(fill_value, tuple)` and explicitly check that it has length 2.